common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/ProductEntitlementMapping.kt

```diff
@@ -10,38 +10,46 @@ data class ProductEntitlementMapping(
         private const val PRODUCTS_KEY = "products"
         private const val PRODUCT_ID_KEY = "id"
         private const val ENTITLEMENTS_KEY = "entitlements"
+        private const val BASE_PLAN_IDENTIFIER = "base_plan_id"
 
         fun fromJson(json: JSONObject): ProductEntitlementMapping {
             val productsArray = json.getJSONArray(PRODUCTS_KEY)
             val mappings = mutableListOf<Mapping>()
             for (productIndex in 0 until productsArray.length()) {
                 val productObject = productsArray.getJSONObject(productIndex)
                 val productIdentifier = productObject.getString(PRODUCT_ID_KEY)
+                val basePlanIdentifier = productObject.getString(BASE_PLAN_IDENTIFIER)
                 val entitlementsArray = productObject.getJSONArray(ENTITLEMENTS_KEY)
                 val entitlements = mutableListOf<String>()
                 for (entitlementIndex in 0 until entitlementsArray.length()) {
                     entitlements.add(entitlementsArray.getString(entitlementIndex))
                 }
-                mappings.add(Mapping(productIdentifier, entitlements))
+                mappings.add(Mapping(productIdentifier, entitlements, basePlanIdentifier))
             }
             return ProductEntitlementMapping(mappings)
         }
     }
 
     data class Mapping(
         val identifier: String,
-        val entitlements: List<String>
+        val entitlements: List<String>,
+        val basePlanIdentifier: String?
     )
 
     fun toMap(): Map<String, List<String>> {
         return mappings.associateBy({ it.identifier }, { it.entitlements })
     }
 
+    fun getMapping(identifier: String): Mapping? {
+        return mappings.find { it.identifier == identifier }
+    }
```

Comment:
This seems unused? Also, we might want to find by both `identifier` and `base_plan_id` first (if it exists) before falling back to the one that has the `identifier` but no `base_plan_id` (the fallback).